api/providers/tasks.py

```diff
@@ -336,44 +287,123 @@ def handle_registration_row(row, initiator, provider, schema, auto_approval=Fals
     auth = Auth(user=initiator)
 
     # Check node
-    node = None
     node_id = metadata.get('Project GUID')
+    node = check_node(node_id, initiator, row, row_title, row_external_id)
+    # Prepare tags
+    tags = metadata.get('Tags', [])
+    # Prepare subjects
+    subject_texts = metadata.get('Subjects', []) or []
+    subject_ids = prepare_subjects(subject_texts, provider, row, row_title, row_external_id)
+    # Prepare institutions
+    institution_names = metadata.get('Affiliated Institutions', []) or []
+    affiliated_institutions = prepare_institutions(institution_names, initiator, row, row_title, row_external_id)
+    # Prepare contributors
+    admin_list = metadata.get('Admin Contributors', []) or []
+    read_only_list = metadata.get('Read-Only Contributors', []) or []
+    read_write_list = metadata.get('Read-Write Contributors', []) or []
+    author_list = metadata.get('Bibliographic Contributors', []) or []
+    parsed_contributors = prepare_contributors(
```

Comment:
Unlike other helper methods that returns a "ready-to-use" object, `prepare_contributors()` returns an instance of `RegistrationBulkUploadContributors` which provides easier permission and citation check/set during `set_draft_contributors()`.